app/gui/src/beforeMain.ts

```diff
@@ -2,8 +2,8 @@
  * @file This module is guaranteed to be imported first, and thus to execute before any
  * other script (including our dependencies).
  */
-
 import * as detect from 'enso-common/src/detect'
+import './config'
```

Comment:
I think config is run in `script` HTML tag, so should not be needed anywhere?

Reply:
Agree, I initially added it because for some reason I had `$config is not defined` error. But it seems like it's gone

Reply:
@farmaazon on the second thought removing isn't a good option either 🤔

Reply:
So I always assumed that the separate <script> tag is needed for bazel. Calling @Frizi in to confirm.

As an alternative (if indeed needed), we could just add import as first in the test file.

Reply:
subscribing to this thread because i also need a hacky workaround for this in one of my PRs:

https://github.com/enso-org/enso/pull/13130/files?w=1#diff-11c61cfd4d9ee97e82ff2eeb497f4d02f56c6e93211d104610f6f528ad681ba0R55-R60

https://github.com/enso-org/enso/blob/dd2a8d314508a833f1fe8a9b7eba9b63b88f0f29/app/gui/src/ReactRoot.tsx#L55-L60

Reply:
> So I always assumed that the separate <script> tag is needed for bazel. Calling @Frizi in to confirm.

It's correct, I'm voting for including the config in setups for tests/storybook. @farmaazon @Frizi @somebody1234 WDYT?

Reply:
The split of config into a separate script is intentional for future work, though right now it is NOT being used in the build process. The intention is to eventually do a post-build config value replacing to guarantee that testing and production builds are exact same, and config is literally the only difference between them. We have a script that does it (`envReplacer.mjs`), but it is currently disabled to keep our build process simple while we still have legacy rust pipeline going. I would however like to keep the config script a separate chunk. All typing issues should be fixed purely on typescript level, since the variable is supposed to be treated as a global `$config` exposed on window before any other application code runs.